src/Shouldly/Internals/StringHelpers.cs

```diff
@@ -170,24 +170,24 @@ internal static string ToSafeString(this char c)
             return c.ToString();
         }
 
-        internal static bool IsNullOrWhiteSpace(this string s)
+        internal static bool IsNullOrWhiteSpace(this string? s)
         {
             return string.IsNullOrWhiteSpace(s);
         }
 
-        internal static string NormalizeLineEndings(this string s)
+        internal static string? NormalizeLineEndings(this string? s)
         {
             return s == null ? null : Regex.Replace(s, @"\r\n?", "\n");
         }
 
-        static string CommaDelimited<T>(this IEnumerable<T> enumerable) where T : class
+        static string CommaDelimited<T>(this IEnumerable<T> enumerable) where T : class?
         {
             return enumerable.DelimitWith(", ");
         }
 
-        static string DelimitWith<T>(this IEnumerable<T> enumerable, string separator) where T : class
+        static string DelimitWith<T>(this IEnumerable<T> enumerable, string? separator) where T : class?
         {
-            return string.Join(separator, enumerable.Select(i => Equals(i, default(T)) ? null : i.ToString()).ToArray());
+            return string.Join(separator, enumerable.Select(i => Equals(i, null) ? null : i.ToString()).ToArray());
```

Comment:
`default(T)` was always `null` already. Using `null` is even nicer now that the other option is `default(T)!`.